File: 11-container-with-most-water/container-with-most-water.c
#define min(a, b) ((a) < (b) ? (a) : (b))
#define max(a, b) ((a) > (b) ? (a) : (b))
int maxArea(int* height, int heightSize) {
        int maxArea = 0;
        int i = 0, j = heightSize - 1;

        while (i < j) {
            int currArea = min(height[i], height[j]) * (j - i);
            maxArea = max(maxArea, currArea);

            if (height[i] < height[j]) {
                i++;
            } else {
                j--;
            }
        }
        return maxArea;
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, a: List[int]) -> int:
        i,j = 0, len(a)-1
        res = 0
        while(i <= j):
            res = max(res,(min(a[i],a[j])*(j-i)))
            if(a[i] < a[j]): i += 1
            else: j -= 1
        return res
